cje-production/dockerfiles/debian/swtgtk3nativebuild/Dockerfile

```
@@ -0,0 +1,42 @@
+FROM debian:10
+
+### user name recognition at runtime w/ an arbitrary uid - for OpenShift deployments
+COPY scripts/uid_entrypoint /usr/local/bin/uid_entrypoint
+RUN chmod u+x /usr/local/bin/uid_entrypoint && \
+    chgrp 0 /usr/local/bin/uid_entrypoint && \
+    chmod g=u /usr/local/bin/uid_entrypoint /etc/passwd
+### end
+
+ENV LANG=en_US.UTF-8
+RUN apt-get update -qq && apt-get install -qq -y \
+      locales \
+      build-essential \
+      procps \
+      git \
+      libgtk-3-dev \
+      freeglut3-dev \
+      webkit2gtk-driver \
+      curl
+
+
+# Use Java17 from JustJ because Debian 10 comes with JDK11 only
+RUN mkdir -p /usr/lib/jvm/justj-17 && \
+      cd /usr/lib/jvm/justj-17 && \
+      curl "https://download.eclipse.org/justj/jres/17/downloads/20230428_1804/org.eclipse.justj.openjdk.hotspot.jre.minimal.stripped-17.0.7-linux-x86_64.tar.gz" | tar -xzf - include/ lib/
```

Wouldn't it be better to get a plain (Temurin)-JDK from the adoptium (API) directly?
See also
- https://api.adoptium.net/
- https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/db246b3b0a4901b4882416962ea99941650cfc1f/JenkinsJobs/AutomatedTests/I_unit_tests.groovy#L137

---

I matched the version that was being used in the Jenkinsfile. The inclusion of the include/ and lib/ here was primarily so I can do local testing of the image.

equinox https://github.com/eclipse-equinox/equinox/blob/5fd2ded80d902fdff53c803dbc2fc7182f75866b/Jenkinsfile#L62

swt: https://github.com/eclipse-platform/eclipse.platform.swt/blob/1ced8a9ca11fe20047d988a0252a9045beeeb070/Jenkinsfile#L61

---

Understand, but the reason JustJ is used in these pipelines is because a stripped down version is provided for download to save bandwidth.
In general that's all an implementation detail and the using pipelines in Equinox and SWT are designed to setup their resources as much as possible and have minimal external requirements so that everything is managed there.
Therefore I would prefer to keep it here as simple as possible.

As far as I could have tested it actually the image doesn't need to contain a JDK at all (as said the pipeline copies the necessary resources anyways).
Therefore I think it's best to remove all about downloading any JDK and setting environment variables for it (the other image could be cleaned-up too IMO).

If you really want to keep it I suggest instead to use:
```suggestion
      curl "https://api.adoptium.net/v3/binary/latest/17/ga/linux/x64/jdk/hotspot/normal/eclipse" | tar -xzf -
```